### ISSUE TITLE
$version is a common variable name and it gets overwritten, if PHP_VERSION_ID doesn't exist

### ISSUE THREAD

**Issue by oittaa** [paragonie/random_compat#71]
It's a corner case, but we should prefix the temporary variable with RandomCompat like we've done with other variables. See #70 586d9ee

**Comment by narfbg** [comment]
Or namespace it and drop 5.2 support ... IMO, nobody who still runs PHP 5.2 would care or even be aware of for random_compat.

**Comment by jdreesen** [comment]
I think 5.2 support is needed for Wordpress.

**Comment by oittaa** [comment]
https://wordpress.org/about/requirements/
> Note: If you are in a legacy environment where you only have older PHP or MySQL versions, WordPress also works with PHP 5.2.4+ and MySQL 5.0+

Of course in the ideal world everybody would be running PHP 7.0 soon after it's released, but that's not going to happen.

**Comment by paragonie-scott** [comment]
> Of course in the ideal world everybody would be running PHP 7.0 soon after it's released, but that's not going to happen.

Indeed, and being a polyfill does not lend well to being bleeding edge.

All of the Paragon projects I've worked on will only run on 5.6 and newer, as far as I'm aware. (We're also working on one that makes heavy use of 7.x features, such as scalar type hinting and strict typing, set to release a week after 7.0 does.)

As much as I'd like to drop 5.2 (and 5.3, and 5.4, and soon to be 5.5), part of the reason we made the code work on 5.2 was precisely to ensure projects like WordPress could adopt it today.
